src/test/javascript/spec/component/short-answer-question/short-answer-question-edit.component.spec.ts

```diff
@@ -0,0 +1,340 @@
+        const mapping1 = new ShortAnswerMapping(spot2, shortAnswerSolution1);
+        const mapping2 = new ShortAnswerMapping(spot3, shortAnswerSolution1);
+        const alternativeMapping = new ShortAnswerMapping(new ShortAnswerSpot(), new ShortAnswerSolution());
+        component.question.correctMappings = [mapping1, mapping2, alternativeMapping];
+        const event = { dragData: shortAnswerSolution1 };
+
+        fixture.detectChanges();
+        component.onDragDrop(spot, event);
+
+        const expectedMapping = new ShortAnswerMapping(spot, shortAnswerSolution1);
+        expect(component.question.correctMappings.pop()).to.deep.equal(expectedMapping);
+        expect(questionUpdatedSpy).to.have.been.calledOnce;
+    });
+
+    it('should setup question editor', () => {
+        component.question.spots = [spot1, spot2];
+        const mapping1 = new ShortAnswerMapping(spot1, shortAnswerSolution1);
+        const mapping2 = new ShortAnswerMapping(spot2, shortAnswerSolution2);
+        component.question.correctMappings = [mapping1, mapping2];
+
+        fixture.detectChanges();
+
+        component.setupQuestionEditor();
+        expect(component.numberOfSpot).to.equal(3);
+        expect(component.optionsWithID).to.deep.equal(['[-option 0]', '[-option 1]']);
+    });
+
+    it('should open', () => {
+        const content = {};
+        const modalService = TestBed.inject(NgbModal);
+        const modalSpy = sinon.spy(modalService, 'open');
+        component.open(content);
+        expect(modalSpy).to.have.been.calledOnce;
+    });
+
+    it('should add spot to cursor', () => {
+        component.numberOfSpot = 2;
+
+        component.addSpotAtCursor();
+        expect(component.numberOfSpot).to.equal(3);
+        expect(component.firstPressed).to.equal(2);
+    });
+
+    it('should add option', () => {
+        component.addOption();
+
+        expect(component.firstPressed).to.equal(2);
+    });
+
+    it('should add and delete text solution', () => {
+        component.question.spots = [spot1, spot2];
+        const mapping1 = new ShortAnswerMapping(spot1, shortAnswerSolution1);
+        const mapping2 = new ShortAnswerMapping(spot2, shortAnswerSolution2);
+        component.question.correctMappings = [mapping1, mapping2];
+
+        component.addTextSolution();
+
+        expect(component.question.solutions!.length).to.equal(3); // 2 -> 3
+
+        component.deleteSolution(shortAnswerSolution2);
+
+        expect(component.question.solutions!.length).to.equal(2); // 3 -> 2
+    });
+
+    it('should add spot at cursor visual mode', () => {
+        const textParts = [['0'], ['0']];
+        const shortAnswerQuestionUtil = TestBed.inject(ShortAnswerQuestionUtil);
+        spyOn(shortAnswerQuestionUtil, 'divideQuestionTextIntoTextParts').and.returnValue(textParts);
+
+        const node = {} as Node;
+
+        const returnValue = ({
+            contains(other: Node | null): boolean {
```

Comment:
`'other' is defined but never used` there are a bunch more warnings we should at least ignore with that  `// tslint ignore` pattern unless we can't fix them

Reply:
👍  55aea5a